src/poisson_numcodecs/poisson.py

```diff
@@ -13,41 +13,115 @@ class Poisson(Codec):
-
-    def decode(self, buf: bytes, out=None) -> np.array:
-        lookup = estimate.make_anscombe_lookup(self.photon_sensitivity)
-        inverse = estimate.make_inverse_lookup(lookup)
-        ndims = int(buf[0])
-        shape = [int(_) for _ in buf[1:ndims+1]]
-        arr = np.frombuffer(buf[ndims+1:], dtype=self.encoded_dtype).reshape(shape)
-        decoded = estimate.lookup(arr, inverse)
-        return decoded.astype(self.decoded_dtype)
+        if self.use_lookup:
+            # produce anscombe lookup_tables
+            input_max = np.iinfo(self.decoded_dtype).max
+            xx = (np.r_[:input_max + 1] - self.dark_signal) / self.photon_sensitivity
+
+            # JEROME: I do not understand why we need to subtract np.sqrt(3/8) here
+            # Also, shouldn't te +3/8 be inside the maximum function as xx is a float?
+            forward = 2.0 / self.beta * (np.sqrt(np.maximum(0, xx) + 3/8) - np.sqrt(3/8))
+            
+            # JEROME : I think this might be better syntax?
+            forward = np.round(forward).astype(self.encoded_dtype)
+            self.forward_table = forward
+
+            _, inverse = np.unique(self.forward_table, return_index=True)
+            inverse += (np.r_[:inverse.size] / 
+                        inverse.size * (inverse[-1] - inverse[-2])/2).astype(self.decoded_dtype)
+            self.inverse_table = inverse
+
+    def _lookup(self, movie, LUT):
+        """
+        Apply lookup table LUT to input movie
+        """
+        return LUT[np.maximum(0, np.minimum(movie, LUT.size-1))]
+
+    def encode(self, buf: np.array):            
+        encoded = np.zeros(buf.shape, dtype=self.encoded_dtype)
+
+        if self.use_lookup:
+            encoded = self._lookup(buf,  self.forward_table)
+        else:
+            # We convert to photons
+            centered = (buf.astype('float') - self.dark_signal) / self.photon_sensitivity
 
+            # https://en.wikipedia.org/wiki/Anscombe_transform for the forward
+            encoded = 2.0 * (np.sqrt(np.maximum(0, centered + 3/8)))
+
+            # This is the part that is applied to discard noise in compressed form
+            encoded = encoded / self.beta 
+
+            # We have to go to integers in a clean way
+            encoded = np.round(encoded)
+        
+        return encoded.astype(self.encoded_dtype)
+    
+    def decode(self, buf, out=None):                   
+        dec = ensure_ndarray(buf).view(self.encoded_dtype)
+ 
+        if self.use_lookup:
+            decoded = self._lookup(dec,  self.inverse_table)
+        else:
+            # We first unapply beta
+            dec = dec.astype('float') * self.beta
+            
+            # https://en.wikipedia.org/wiki/Anscombe_transform for the inverse without bias
+            dec = dec**2 / 4.0 - 1/8
+
+            # We convert back to arbitrary pixels
+            dec = dec * self.photon_sensitivity + self.dark_signal
+
+            # We have to go back to integers
```

Comment:
The code explains itself.
```suggestion
            # We have to go back to integers
```